app/scripts/controllers/threebox.js

```diff
@@ -42,7 +41,9 @@ class ThreeBoxController {
         }
       },
       processPersonalMessage: (msgParams) => {
-        return Promise.resolve(signPersonalMessage(msgParams))
+        return Promise.resolve(keyringController.signPersonalMessage(msgParams, {
+          withAppKeyOrigin: '3box.metamask.io',
```

Comment:
Shouldn't this be prefixed with `https://`?

Reply:
maybe, why do you say so?

Reply:
Without the scheme, there's no way to distinguish it from other protocols, so we effectively disallow other protocols.

The scheme was included in @danfinlay 's description of this feature in the EIP thread: https://ethereum-magicians.org/t/eip-erc-app-keys-application-specific-wallet-accounts/2742/3

Reply:
Yeah, I agree it should have a prefix of some sort. Strangely this is a decent opportunity for a non-existent prefix, to ensure no DNS attack could impersonate this domain. For example: `browser-extension://3box.metamask.io`.

Or, maybe it would be crazy, but we could even pick a domain that is friendly towards portability between wallets. Something like `wallet://backup` or something...